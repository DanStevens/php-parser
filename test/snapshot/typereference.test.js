const parser = require('../main');

describe("typereference", function() {
  it("int", function() {
    expect(parser.parseEval('function fn(int $arg) {}')).toMatchSnapshot();
  });
  it("int (uppercase)", function() {
    expect(parser.parseEval('function fn(INT $arg) {}')).toMatchSnapshot();
  });
  it("float", function() {
    expect(parser.parseEval('function fn(float $arg) {}')).toMatchSnapshot();
  });
  it("float (uppercase)", function() {
    expect(parser.parseEval('function fn(FLOAT $arg) {}')).toMatchSnapshot();
  });
  it("bool", function() {
    expect(parser.parseEval('function fn(bool $arg) {}')).toMatchSnapshot();
  });
  it("bool (uppercase)", function() {
    expect(parser.parseEval('function fn(BOOL $arg) {}')).toMatchSnapshot();
  });
  it("string", function() {
    expect(parser.parseEval('function fn(string $arg) {}')).toMatchSnapshot();
  });
  it("string (uppercase)", function() {
    expect(parser.parseEval('function fn(STRING $arg) {}')).toMatchSnapshot();
  });
  it("array", function() {
    expect(parser.parseEval('function fn(array $arg) {}')).toMatchSnapshot();
  });
  it("array (uppercase)", function() {
    expect(parser.parseEval('function fn(ARRAY $arg) {}')).toMatchSnapshot();
  });
  it("callable", function() {
    expect(parser.parseEval('function fn(callable $arg) {}')).toMatchSnapshot();
  });
  it("callable (uppercase)", function() {
    expect(parser.parseEval('function fn(CALLABLE $arg) {}')).toMatchSnapshot();
  });
  it("object", function() {
    expect(parser.parseEval('function fn(object $arg) {}')).toMatchSnapshot();
  });
  it("object (uppercase)", function() {
    expect(parser.parseEval('function fn(OBJECT $arg) {}')).toMatchSnapshot();
  });
  it("iterable", function() {
    expect(parser.parseEval('function fn(iterable $arg) {}')).toMatchSnapshot();
  });
  it("iterable (uppercase)", function() {
    expect(parser.parseEval('function fn(ITERABLE $arg) {}')).toMatchSnapshot();
  });
  it("class", function() {
    expect(parser.parseEval('function fn(Foo $arg) {}')).toMatchSnapshot();
  });
  it("class (2)", function() {
    expect(parser.parseEval('function fn(Foo\\Foo $arg) {}')).toMatchSnapshot();
  });
});
