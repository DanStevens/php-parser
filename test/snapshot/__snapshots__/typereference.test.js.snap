// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typereference array (uppercase) 1`] = `
Program {
  "children": Array [
    _Function {
      "arguments": Array [
        Parameter {
          "byref": false,
          "kind": "parameter",
          "name": "arg",
          "nullable": false,
          "type": TypeReference {
            "kind": "typereference",
            "name": "array",
            "raw": "ARRAY",
          },
          "value": null,
          "variadic": false,
        },
      ],
      "body": Block {
        "children": Array [],
        "kind": "block",
      },
      "byref": false,
      "kind": "function",
      "name": Identifier {
        "kind": "identifier",
        "name": "fn",
      },
      "nullable": false,
      "type": null,
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`typereference array 1`] = `
Program {
  "children": Array [
    _Function {
      "arguments": Array [
        Parameter {
          "byref": false,
          "kind": "parameter",
          "name": "arg",
          "nullable": false,
          "type": TypeReference {
            "kind": "typereference",
            "name": "array",
            "raw": "array",
          },
          "value": null,
          "variadic": false,
        },
      ],
      "body": Block {
        "children": Array [],
        "kind": "block",
      },
      "byref": false,
      "kind": "function",
      "name": Identifier {
        "kind": "identifier",
        "name": "fn",
      },
      "nullable": false,
      "type": null,
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`typereference bool (uppercase) 1`] = `
Program {
  "children": Array [
    _Function {
      "arguments": Array [
        Parameter {
          "byref": false,
          "kind": "parameter",
          "name": "arg",
          "nullable": false,
          "type": TypeReference {
            "kind": "typereference",
            "name": "bool",
            "raw": "BOOL",
          },
          "value": null,
          "variadic": false,
        },
      ],
      "body": Block {
        "children": Array [],
        "kind": "block",
      },
      "byref": false,
      "kind": "function",
      "name": Identifier {
        "kind": "identifier",
        "name": "fn",
      },
      "nullable": false,
      "type": null,
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`typereference bool 1`] = `
Program {
  "children": Array [
    _Function {
      "arguments": Array [
        Parameter {
          "byref": false,
          "kind": "parameter",
          "name": "arg",
          "nullable": false,
          "type": TypeReference {
            "kind": "typereference",
            "name": "bool",
            "raw": "bool",
          },
          "value": null,
          "variadic": false,
        },
      ],
      "body": Block {
        "children": Array [],
        "kind": "block",
      },
      "byref": false,
      "kind": "function",
      "name": Identifier {
        "kind": "identifier",
        "name": "fn",
      },
      "nullable": false,
      "type": null,
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`typereference callable (uppercase) 1`] = `
Program {
  "children": Array [
    _Function {
      "arguments": Array [
        Parameter {
          "byref": false,
          "kind": "parameter",
          "name": "arg",
          "nullable": false,
          "type": TypeReference {
            "kind": "typereference",
            "name": "callable",
            "raw": "CALLABLE",
          },
          "value": null,
          "variadic": false,
        },
      ],
      "body": Block {
        "children": Array [],
        "kind": "block",
      },
      "byref": false,
      "kind": "function",
      "name": Identifier {
        "kind": "identifier",
        "name": "fn",
      },
      "nullable": false,
      "type": null,
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`typereference callable 1`] = `
Program {
  "children": Array [
    _Function {
      "arguments": Array [
        Parameter {
          "byref": false,
          "kind": "parameter",
          "name": "arg",
          "nullable": false,
          "type": TypeReference {
            "kind": "typereference",
            "name": "callable",
            "raw": "callable",
          },
          "value": null,
          "variadic": false,
        },
      ],
      "body": Block {
        "children": Array [],
        "kind": "block",
      },
      "byref": false,
      "kind": "function",
      "name": Identifier {
        "kind": "identifier",
        "name": "fn",
      },
      "nullable": false,
      "type": null,
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`typereference class (2) 1`] = `
Program {
  "children": Array [
    _Function {
      "arguments": Array [
        Parameter {
          "byref": false,
          "kind": "parameter",
          "name": "arg",
          "nullable": false,
          "type": ClassReference {
            "kind": "classreference",
            "name": "Foo\\\\Foo",
            "resolution": "qn",
          },
          "value": null,
          "variadic": false,
        },
      ],
      "body": Block {
        "children": Array [],
        "kind": "block",
      },
      "byref": false,
      "kind": "function",
      "name": Identifier {
        "kind": "identifier",
        "name": "fn",
      },
      "nullable": false,
      "type": null,
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`typereference class 1`] = `
Program {
  "children": Array [
    _Function {
      "arguments": Array [
        Parameter {
          "byref": false,
          "kind": "parameter",
          "name": "arg",
          "nullable": false,
          "type": ClassReference {
            "kind": "classreference",
            "name": "Foo",
            "resolution": "uqn",
          },
          "value": null,
          "variadic": false,
        },
      ],
      "body": Block {
        "children": Array [],
        "kind": "block",
      },
      "byref": false,
      "kind": "function",
      "name": Identifier {
        "kind": "identifier",
        "name": "fn",
      },
      "nullable": false,
      "type": null,
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`typereference float (uppercase) 1`] = `
Program {
  "children": Array [
    _Function {
      "arguments": Array [
        Parameter {
          "byref": false,
          "kind": "parameter",
          "name": "arg",
          "nullable": false,
          "type": TypeReference {
            "kind": "typereference",
            "name": "float",
            "raw": "FLOAT",
          },
          "value": null,
          "variadic": false,
        },
      ],
      "body": Block {
        "children": Array [],
        "kind": "block",
      },
      "byref": false,
      "kind": "function",
      "name": Identifier {
        "kind": "identifier",
        "name": "fn",
      },
      "nullable": false,
      "type": null,
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`typereference float 1`] = `
Program {
  "children": Array [
    _Function {
      "arguments": Array [
        Parameter {
          "byref": false,
          "kind": "parameter",
          "name": "arg",
          "nullable": false,
          "type": TypeReference {
            "kind": "typereference",
            "name": "float",
            "raw": "float",
          },
          "value": null,
          "variadic": false,
        },
      ],
      "body": Block {
        "children": Array [],
        "kind": "block",
      },
      "byref": false,
      "kind": "function",
      "name": Identifier {
        "kind": "identifier",
        "name": "fn",
      },
      "nullable": false,
      "type": null,
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`typereference int (uppercase) 1`] = `
Program {
  "children": Array [
    _Function {
      "arguments": Array [
        Parameter {
          "byref": false,
          "kind": "parameter",
          "name": "arg",
          "nullable": false,
          "type": TypeReference {
            "kind": "typereference",
            "name": "int",
            "raw": "INT",
          },
          "value": null,
          "variadic": false,
        },
      ],
      "body": Block {
        "children": Array [],
        "kind": "block",
      },
      "byref": false,
      "kind": "function",
      "name": Identifier {
        "kind": "identifier",
        "name": "fn",
      },
      "nullable": false,
      "type": null,
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`typereference int 1`] = `
Program {
  "children": Array [
    _Function {
      "arguments": Array [
        Parameter {
          "byref": false,
          "kind": "parameter",
          "name": "arg",
          "nullable": false,
          "type": TypeReference {
            "kind": "typereference",
            "name": "int",
            "raw": "int",
          },
          "value": null,
          "variadic": false,
        },
      ],
      "body": Block {
        "children": Array [],
        "kind": "block",
      },
      "byref": false,
      "kind": "function",
      "name": Identifier {
        "kind": "identifier",
        "name": "fn",
      },
      "nullable": false,
      "type": null,
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`typereference iterable (uppercase) 1`] = `
Program {
  "children": Array [
    _Function {
      "arguments": Array [
        Parameter {
          "byref": false,
          "kind": "parameter",
          "name": "arg",
          "nullable": false,
          "type": TypeReference {
            "kind": "typereference",
            "name": "iterable",
            "raw": "ITERABLE",
          },
          "value": null,
          "variadic": false,
        },
      ],
      "body": Block {
        "children": Array [],
        "kind": "block",
      },
      "byref": false,
      "kind": "function",
      "name": Identifier {
        "kind": "identifier",
        "name": "fn",
      },
      "nullable": false,
      "type": null,
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`typereference iterable 1`] = `
Program {
  "children": Array [
    _Function {
      "arguments": Array [
        Parameter {
          "byref": false,
          "kind": "parameter",
          "name": "arg",
          "nullable": false,
          "type": TypeReference {
            "kind": "typereference",
            "name": "iterable",
            "raw": "iterable",
          },
          "value": null,
          "variadic": false,
        },
      ],
      "body": Block {
        "children": Array [],
        "kind": "block",
      },
      "byref": false,
      "kind": "function",
      "name": Identifier {
        "kind": "identifier",
        "name": "fn",
      },
      "nullable": false,
      "type": null,
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`typereference object (uppercase) 1`] = `
Program {
  "children": Array [
    _Function {
      "arguments": Array [
        Parameter {
          "byref": false,
          "kind": "parameter",
          "name": "arg",
          "nullable": false,
          "type": TypeReference {
            "kind": "typereference",
            "name": "object",
            "raw": "OBJECT",
          },
          "value": null,
          "variadic": false,
        },
      ],
      "body": Block {
        "children": Array [],
        "kind": "block",
      },
      "byref": false,
      "kind": "function",
      "name": Identifier {
        "kind": "identifier",
        "name": "fn",
      },
      "nullable": false,
      "type": null,
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`typereference object 1`] = `
Program {
  "children": Array [
    _Function {
      "arguments": Array [
        Parameter {
          "byref": false,
          "kind": "parameter",
          "name": "arg",
          "nullable": false,
          "type": TypeReference {
            "kind": "typereference",
            "name": "object",
            "raw": "object",
          },
          "value": null,
          "variadic": false,
        },
      ],
      "body": Block {
        "children": Array [],
        "kind": "block",
      },
      "byref": false,
      "kind": "function",
      "name": Identifier {
        "kind": "identifier",
        "name": "fn",
      },
      "nullable": false,
      "type": null,
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`typereference string (uppercase) 1`] = `
Program {
  "children": Array [
    _Function {
      "arguments": Array [
        Parameter {
          "byref": false,
          "kind": "parameter",
          "name": "arg",
          "nullable": false,
          "type": TypeReference {
            "kind": "typereference",
            "name": "string",
            "raw": "STRING",
          },
          "value": null,
          "variadic": false,
        },
      ],
      "body": Block {
        "children": Array [],
        "kind": "block",
      },
      "byref": false,
      "kind": "function",
      "name": Identifier {
        "kind": "identifier",
        "name": "fn",
      },
      "nullable": false,
      "type": null,
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`typereference string 1`] = `
Program {
  "children": Array [
    _Function {
      "arguments": Array [
        Parameter {
          "byref": false,
          "kind": "parameter",
          "name": "arg",
          "nullable": false,
          "type": TypeReference {
            "kind": "typereference",
            "name": "string",
            "raw": "string",
          },
          "value": null,
          "variadic": false,
        },
      ],
      "body": Block {
        "children": Array [],
        "kind": "block",
      },
      "byref": false,
      "kind": "function",
      "name": Identifier {
        "kind": "identifier",
        "name": "fn",
      },
      "nullable": false,
      "type": null,
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;
